layer/trace_layer.cpp

```diff
@@ -376,16 +376,20 @@ VKAPI_ATTR VkResult VKAPI_CALL EnumerateDeviceExtensionProperties(VkPhysicalDevi
         }
 
         std::vector<VkExtensionProperties> downstream_properties(downstream_property_count);
-        result = instance_table->EnumerateDeviceExtensionProperties(
-            wrapped_device, pLayerName, &downstream_property_count, &downstream_properties[0]);
+        if (downstream_property_count != 0)
```

Comment:
For this check to be required I think there would have to be a bug in another layer, the loader (possibly), or the Vulkan implementation. Can you provide more details on what other layers you are using, what Vulkan driver, and what platform?

Reply:
Thank you for your PR!  

I agree with Andrew - my opinion is that passing 0 as `downstream_property_count` should not cause a crash.

The spec says:
> If pPropertyCount is less than the number of extension properties available, at most pPropertyCount structures will be written, and VK_INCOMPLETE will be returned instead of VK_SUCCESS, to indicate that not all the available properties were returned.

So at most "0" structures will be written, which means `EnumerateDeviceExtensionProperties` should write nothing and return `VK_INCOMPLETE`.

However, on gcc at least (https://godbolt.org/z/noPq1W8ra) I think it's possible that `&downstream_properties[0]` will be `nullptr` if the vector is constructed with 0 size.  So this call to `EnumerateDeviceExtensionProperties` should actually behave like we're querying the count and store the count in `*downstream_property_count`.  That isn't what I expected here but is harmless and should not crash.

Since this PR is simple and removes a function call we probably don't want, I may suggest to the team that we accept the PR, but I want to understand more about the context here.